tests/data/test_AtomicData.py

```diff
@@ -19,6 +19,30 @@ def test_from_ase(CuFcc):
         assert data[key].shape == (len(atoms), 3)  # 4 species in this atoms
 
 
+def test_to_ase(CH3CHO):
+    atoms, data = CH3CHO
+    to_ase_atoms = data.to_ase()
+    assert np.allclose(atoms.get_positions(), to_ase_atoms.get_positions())
+    assert np.array_equal(atoms.get_atomic_numbers(), to_ase_atoms.get_atomic_numbers())
+    assert np.array_equal(atoms.get_pbc(), to_ase_atoms.get_pbc())
+    assert np.array_equal(atoms.get_cell(), to_ase_atoms.get_cell())
+
+
+def test_to_ase_batches(atomic_batch):
+    atomic_data = AtomicData.from_dict(vars(atomic_batch))
+    to_ase_atoms_batch = atomic_data.to_ase()
+    for batch_idx, atoms in enumerate(to_ase_atoms_batch):
+        mask = atomic_data.batch == batch_idx
+        assert atoms.get_positions().shape == (len(atoms), 3)
+        assert np.allclose(atoms.get_positions(), atomic_data.pos[mask])
+        assert atoms.get_atomic_numbers().shape == (len(atoms),)
+        assert np.array_equal(
+            atoms.get_atomic_numbers(), atomic_data.atomic_numbers[mask]
+        )
+        assert np.array_equal(atoms.get_cell(), atomic_data.cell[batch_idx])
+        assert np.array_equal(atoms.get_pbc(), atomic_data.pbc[batch_idx])
```

Comment:
It looks like `ase.Atoms` basically implements these checks in its own magic `__eq__` method, which is called if you do something like `assert atoms1 == atoms2`. (See https://gitlab.com/ase/ase/-/blob/master/ase/atoms.py#L1888)

Maybe cleaner to use that? Save some lines. (For the other test too.)

Reply:
I see, so is there a way to create an `ase.Atoms` object directly from the `Batch` object (or I guess how do I get the second `ase.Atoms` object in this test?)

Reply:
So when I try this for the `CH3CHO` fixture, I get this:

test_AtomicData.py::test_to_ase[float32] FAILED                          [ 50%]
tests\data\test_AtomicData.py:22 (test_to_ase[float32])
Atoms(symbols='OCHCH3', pbc=False) != Atoms(symbols='OCHCH3', pbc=False)

Expected :Atoms(symbols='OCHCH3', pbc=False)
Actual   :Atoms(symbols='OCHCH3', pbc=False)

The only difference between the two `ase.Atoms` objects is the precision of the atomic positions, so would that be the reason for this error (since the method in the gitlab code seems to check for equality)?

Reply:
aha yes that's a good catch @albertzhu01 , given that, I'd go back to your original code.